From d3a2453aea167976f6496436182a74b6b7917df7 Mon Sep 17 00:00:00 2001
From: Darby Johnston <darbyjohnston@yahoo.com>
Date: Sat, 7 Jun 2025 14:39:13 -0700
Subject: [PATCH] Remove unused variables

---
 tests/otiotest.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/tests/otiotest.cpp b/tests/otiotest.cpp
index 90051fbd45..c75d5fd3b2 100644
--- a/tests/otiotest.cpp
+++ b/tests/otiotest.cpp
@@ -120,7 +120,6 @@ TEST_CASE("Export/import", "[OTIO]")
         QString inputPath = sourcesPath + "/dataset/test-guides-markers.otio";
         OTIO_NS::SerializableObject::Retainer<OTIO_NS::Timeline> otioTestTimeline(
             dynamic_cast<OTIO_NS::Timeline *>(OTIO_NS::Timeline::from_json_file(inputPath.toStdString())));
-        const int otioClipCount = otioTestTimeline->find_clips().size();
 
         // Import the OTIO test timeline.
         OtioImport otioImport;
@@ -199,7 +198,6 @@ TEST_CASE("Export/import", "[OTIO]")
         OTIO_NS::SerializableObject::Retainer<OTIO_NS::Timeline> otioTestTimeline(
             dynamic_cast<OTIO_NS::Timeline *>(OTIO_NS::Timeline::from_json_file(inputPath.toStdString())));
         const auto otioClips = otioTestTimeline->find_clips();
-        const int otioClipCount = otioClips.size();
 
         // Import the OTIO test timeline.
         OtioImport otioImport;
@@ -250,7 +248,6 @@ TEST_CASE("Export/import", "[OTIO]")
         OTIO_NS::SerializableObject::Retainer<OTIO_NS::Timeline> otioTestTimeline(
             dynamic_cast<OTIO_NS::Timeline *>(OTIO_NS::Timeline::from_json_file(inputPath.toStdString())));
         const auto otioClips = otioTestTimeline->find_clips();
-        const int otioClipCount = otioClips.size();
 
         // Import the OTIO test timeline.
         OtioImport otioImport;
-- 
GitLab

