--- kdenlive-20.04.2/data/org.kde.kdenlive.appdata.xml	2020-06-09 17:22:41.000000000 -0400
+++ kdenlive-20.04.2/data/org.kde.kdenlive.appdata.xml-new	2020-07-06 19:15:55.156144232 -0400
@@ -267,6 +267,7 @@
     </ul>
   </description>
   <releases>
+    <release version="20.04.3" date="2020-07-09"/>
     <release date="2020-05-11" version="20.04.1"/>
   </releases>
   <url type="homepage">https://kdenlive.org/</url>
